sdk/storage/Azure.Storage.Blobs.DataMovement/tests/BlobScannerTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs;
+using Azure.Core.TestFramework;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs.Specialized;
+using Azure.Storage.Blobs.DataMovement;
+using NUnit.Framework;
+
+namespace Azure.Storage.Tests
+{
+    public class BlobScannerTests : BlobDataMovementTests
+    {
+        public BlobScannerTests(bool async, BlobClientOptions.ServiceVersion serviceVersion)
+            : base(async, serviceVersion, RecordedTestMode.Record /* RecordedTestMode.Record /* to re-record */)
+        {
+        }
+
+        [RecordedTest]
+        public async Task ScanBlobContainer()
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            PageBlobClient blob = await CreatePageBlobClientAsync(test.Container, 4 * Constants.KB);
+            string blobName = blob.Name;
+
+            var data = GetRandomBuffer(Constants.KB);
+
+            using (var stream = new MemoryStream(data))
+            {
+                await blob.UploadPagesAsync(
+                    content: stream,
+                    offset: Constants.KB);
+            }
```

Comment:
I took this from the PageBlobClientTests, but I'm confused on what this does.

Does this make a blob 4kb in size, and then upload the 1kb of random data to the blob, with a 1 kb offset from the blob start? So looking like this on the server:
<---- 1 kb free ----><---- random data ----><---------------- 2kb free ---------------->

Also does the server actually create the blob at line 32, or does it only commit and become visible to .ScanBlobsAsync() after uploading data? (I'll try it, might not even need the upload in this test)

Reply:
Test passes without uploading on NET 2.1, but fails on 5.0 and 4.6.1... I'll leave the upload, I guess.

Reply:
Yes that helper function actually creates the blob at line 32 and is visible already.

What's `ScanBlobsAsync()` ?

Does the blob not show up unless you upload?

Reply:
Do you mean the ScanBlobContainerAsync() test or the GetBlobsAsync() from inside the scanner?

I'm not sure; when I upload, tests pass on all platforms, but they fail on 5.0/4.6.1 if I don't. The error says "this container is being deleted".

Reply:
... you're the one who literally said ScanBlobsAsync(), that's why I'm asking you to clarify. I will assume you're asking about GetBlobAsync(), but that doesn't matter.

If the error you're getting is "The specified container is being deleted." looks like for some reason the disposing container is using the same name. Maybe its using the same random seed to create the containers in the next tests. You could either just wait a while between running tests between 5.0 and 4.6.1. I think if it passes in one place, this is fine, just wait a while and run 5.0 or 4.6.1 later just to make sure everything is running fine.

Reply:
Oh. My head is screwed on backwards. I meant .GetBlobsSync(), sorry. I'll try doing the tests with a delay then.